pandas/_testing.py

```diff
@@ -1038,7 +1038,8 @@ def assert_extension_array_equal(
 
     if hasattr(left, "asi8") and type(right) == type(left):
         # Avoid slow object-dtype comparisons
-        assert_numpy_array_equal(left.asi8, right.asi8)
+        # np.asarray for case where we have a np.MaskedArray
+        assert_numpy_array_equal(np.asarray(left.asi8), np.asarray(right.asi8))
```

Comment:
How can `asi8` ever return a MaskedArray?

Reply:
we have one test where we pass a MaskedArray to the constructor (I think DTI)

Reply:
> we have one test where we pass a MaskedArray to the constructor (I think DTI)

But shouldn't that be converted to a normal array upon construction?

Reply:
id have no problem with enforcing this, but AFAIK we dont at the moment